src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1684,6 +1684,17 @@ internal Solution WithNewWorkspace(Workspace workspace, int workspaceVersion)
             return new Solution(newState);
         }
 
+        internal Solution WithProjectCommandLineOptions(ProjectId projectId, string? commandLineOptions)
```

Comment:
Can we mark these as [Obsolete] with the usual comment we do for F# internal APIs so that way it's clear this is only in support of F#? We don't want other people using this...